lib/Sema/CSSimplify.cpp

```diff
@@ -1600,15 +1606,6 @@ ConstraintSystem::matchTypesBindTypeVar(
     }
   }
 
-  // Check whether the type variable must be bound to a materializable
-  // type.
-  if (typeVar->getImpl().mustBeMaterializable()) {
```

Comment:
@xedin Note the old behavior here was kind of weird and used the word 'materializable' that I don't like:

- CanBindToLValue meant can bind to LValueType *or* InOutType
- CanBindToInOut meant can bind to InOutType

I don't think this was intended. I fixed it but I don't think there's a behavior change in practice since the only time we see InOutType in type variables is with BindParam constraint.